packages/api/src/controllers/webhook.ts

```diff
@@ -241,28 +241,57 @@ app.post(
   async (req, res) => {
     const { id } = req.params;
     const webhook = await db.webhook.get(id);
-    if (!webhook) {
-      return res.status(404).json({ errors: ["not found"] });
+    if (!webhook || webhook.deleted) {
+      return res.status(404).json({ errors: ["webhook not found or deleted"] });
     }
 
-    const doc = req.body.status;
-    const webhookResponse = req.body.response;
+    const { response, errorMessage } = req.body as WebhookStatusPayload;
 
-    if (!doc) {
-      return res.status(422).json({ errors: ["missing status in payload"] });
+    if (!response || !response.response) {
+      return res.status(422).json({ errors: ["missing response in payload"] });
+    }
+    if (!response.response.body && response.response.body !== "") {
+      return res
+        .status(400)
+        .json({ errors: ["missing body in payload response"] });
     }
 
-    await db.webhook.update(req.params.id, {
-      ...webhook,
-      status: doc,
-    });
-
-    if (webhookResponse) {
-      webhookResponse.id = uuid();
-      let w: WithID<WebhookResponse> = webhookResponse;
-      await db.webhookResponse.create(w);
+    try {
+      const triggerTime = response.createdAt ?? Date.now();
+      let status: DBWebhook["status"] = { lastTriggeredAt: triggerTime };
+      if (
+        response.statusCode >= 300 ||
+        !response.statusCode ||
+        response.statusCode === 0
+      ) {
+        status = {
+          ...status,
+          lastFailure: {
+            error: errorMessage,
+            timestamp: triggerTime,
+            statusCode: response.statusCode,
+            response: response.response.body,
+          },
+        };
+      }
+      await db.webhook.updateStatus(webhook.id, status);
+    } catch (e) {
+      console.log(
+        `Unable to store status of webhook ${webhook.id} url: ${webhook.url}`
+      );
     }
 
+    await db.webhookResponse.create({
+      id: uuid(),
+      webhookId: webhook.id,
+      createdAt: Date.now(),
+      statusCode: response.statusCode,
+      response: {
+        body: response.response.body,
+        status: response.statusCode,
```

Comment:
There is also a lot more stuff in the response object. Please save at least the `response.headers` here as well but do go through the schema to figure out anything else we want to record. Ideally we should at least keep it consistent with the webhook cannon code, which right now it still isnt'.